### ISSUE TITLE
Encoding Problem - Windows Console

### ISSUE THREAD

**Issue by nilsmarkmeyer** [tomakehurst/wiremock#310]
If I run wiremock in windows in maven there is an encoding problem.

Wiremock is used for our tests. We set our maven encoding to "UTF-8". (`set MAVEN_OPTS= -Dfile.encoding="UTF-8"`)
I think wiremock will start without that encoding parameter, so it trys to convert the current UTF-8 encoded file again, because it use the standard windows encoding.

Is it possible setting the encoding parameter of files while wiremock configuration?

When I run the test in the IDE there is no problem (Because the Encoding matchs). Just if I run them in the console there are errors.

Here the jUnit Error:

```
org.junit.ComparisonFailure: expected:<...------
{
  "text": "[üöäåÅש؁ж໗🛀⟰п]"
}
----------------...> but was:<...------
{
  "text": "[Ã¼Ã¶Ã¤Ã¥Ã…×©Ø�Ð¶à»—ðŸ›€âŸ°Ð¿]"
}
```

**Comment by tomakehurst** [comment]
WireMock is hard-coded to UTF-8 I'm afraid. Might be better to convert your files to UTF-8.

**Comment by nilsmarkmeyer** [comment]
Our files are UTF-8 encoded. I guess wiremock will be started with the iso-code converts the files twice.

**Comment by nilsmarkmeyer** [comment]
The files are UTF-8 encoded. Maybe the Tests in Maven will run Wiremock with the windows encoding, so wiremock "think" the file is not utf-8 encoded, so the file will be double utf-8 encoded or something like that. If the files wouldn't be encoded the direct tests via `JAVA` command would end in error too.

**Comment by tomakehurst** [comment]
WireMock forces to UTF-8 when reading files so I wouldn't have thought it would matter what the file.encoding property was set to.

Not totally sure I understand your last statement, did setting the property explicitly fix your problem?

**Comment by nilsmarkmeyer** [comment]
Ok - I try to explain it again:

I use wiremock for the unit-tests. If I run the single test via `java` and encoding parameter it works. If I run `mvn verify` in my windows console there are encoding errors. At the Mac / Unix console works all fine. 

Now I fixed the windows console problem with the `MVN_OPTS` Parameter, so the expected result in unit test is fine. But the return value of wiremock is now incorrect. 

The files are all UTF-8 Encoded.

I think, when wiremock start at the maven verify command it try to convert all files to UTF-8 again. Is there any parameter to configure this or do you have another idea why this could happen?

**Comment by arcuri82** [comment]
Hi,
  I ended up in the same situation: tests running perfectly fine on IDE (IntelliJ), but when using `mvn verify` on Windows I got all kinds of encoding issues, as I have some special non-ASCII characters in the WireMock responses. However, one difference from @nilsmarkmeyer is that I am not reading any file: I encode the WireMock response directly as a Java string in my test cases. Somehow, when run from Maven, WireMock sends the http response in ISO-8859-1 instead of UTF-8, and that does crash the XML parser on the other side (ie the system I am testing) as it expects UTF-8. I tried to debug what is going on, but unfortunately with no result :(  in my case, my workaround was to get rid of all the non-ASCII characters (was just one) in my tests, as anyway I did not really need them

**Comment by tomakehurst** [comment]
If you could post your build and test code (or at least examples) that would be incredibly helpful.

Incidentally, have you only tried it on Windows, or is this a problem running from Maven generally?

**Comment by arcuri82** [comment]
unfortunately, it is proprietary code that I cannot share :(  I only tried from Windows, so not sure if it is a general issue with Maven. I can try to reproduce the issue in a standalone project, once I manage to find some time for it :) however, I just wanted to report it as it looks like the issue was independent on whether the response body is loaded from file or directly encoded in the Java tests

**Comment by tomakehurst** [comment]
OK, I'll take a look when I can.

**Comment by 0xqab** [comment]
I had the same issue and could fix it by explicitly setting UTF-8 encoding in the Maven Surefire plugin.
```
<plugin>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <argLine>-Dfile.encoding=UTF-8</argLine>
    </configuration>
</plugin>
```

**Comment by sylvainlaurent** [comment]
I think the problem is in `com.github.tomakehurst.wiremock.common.Strings.bytesFromString(String)` where the call to `str.getBytes();` depends on the current platform encoding.
The correct call should `str.getBytes(UTF_8);`.

**Comment by TheJavaGuy** [comment]
This seems to be fixed in 2.10.1 (and possibly in some earlier version too). Maybe this issue should be closed?
